arkouda/strings.py

```diff
@@ -91,14 +184,58 @@ def binop(self, other, op):
                                                               self.objtype,
                                                               json.dumps([other]))
         else:
-            raise ValueError("Strings: {} not supported between Strings and {}".format(op, type(other)))
+            raise ValueError("Strings: {} not supported between Strings and {}"\
+                             .format(op, type(other)))
         repMsg = generic_msg(msg)
         return create_pdarray(repMsg)
 
-    def __eq__(self, other):
-        return self.binop(other, "==")
+    def __eq__(self, other) -> bool:
+        """
+        Indicates whether the instance is equal to the other object
+        
+        Parameters
+        ----------
+        other : object
+            The object to be compared to the current instance
+        
+        Returns
+        -------
+        bool
+            boolean indicating whether the instance is equal to the 
+            other object
+        
+        Raises
+        ------
+        RuntimeError
+            Raised if there is a server-side error thrown in the 
+            execution of the binop('==') method
+        """
+        if not isinstance(other, Strings):
```

Comment:
Ah, this is a good question. In my experience, it's customary to return False in the __eq__ method if the object being compared to is of a different type. Here's an example from the Faust [Join](https://github.com/robinhood/faust/blob/68836cb18b35019c382bde30a277347a0b885c9e/faust/joins.py) class. 

![image](https://user-images.githubusercontent.com/10785153/90766089-004b8d80-e2b9-11ea-887e-029df6537320.png)

But, if you feel strongly otherwise, definitely okay with raising an error instead.

Reply:
Actually, let's remove the type check and revert `__eq__` back to what it was before the PR. I forgot, we need to support scalar comparisons as well, e.g. `ak.Strings == "mystr"`. Better to let `binop` handle the type checking.

Reply:
Cool, okay, will be in next commit